src/NServiceBus.Core/Causation/ConversationIdStrategyContext.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace NServiceBus
+{
+    using System.Collections.Generic;
+    using Pipeline;
+
+    /// <summary>
+    /// Provides context when generating message conversation IDs.
+    /// </summary>
+    public class ConversationIdStrategyContext
+    {
+        /// <summary>
+        /// Creates a new context.
+        /// </summary>
+        public ConversationIdStrategyContext(OutgoingLogicalMessage message, IReadOnlyDictionary<string, string> headers)
```

Comment:
@andreasohlund One last question: Should we be guarding against `headers` being null as well, or is there a reason to allow null headers?

Reply:
I just missed it, we should add a guard